### PR TITLE
[wcchat] Telnet channel actor

### PR DESCRIPTION
**Pull request by brooksmtownsend**
Initial working IMPL of the telnet channel actor

### CODE REVIEW

**Review by autodidaddict — APPROVED** [review]
suggestion for commands we should support:
* `/who` - lists all the people who are online (obtained from presence) - this should show everyone, not just the people online via telnet
* `/msg (person) (text)` - send a DM to the given person
* `/join (room)` - joins a room
* `/leave (room)` - leaves a room
* `/rooms` - shows the list of rooms

File: wasmcloud-chat/actors/telnet-channel/src/lib.rs [hunk 1]
```diff
@@ -136,38 +161,60 @@ fn receive_text(session: String, message_text: String) -> HandlerResult<bool> {
...
+    debug!(
+        "command {} received from user {}",
+        command, origin_session.username
+    );
+    let telnet_host = telnet::default();
+    let res = match command {
+        HELP_CMD => telnet_host.send_text(origin_session.id.clone(), HELP_TEXT.to_string()),
+        ROOMS_CMD => Ok(true),
+        WHO_CMD => Ok(true),
+        _ if command.starts_with(MSG_CMD) => {
```
Comment (autodidaddict) [inline wasmcloud-chat/actors/telnet-channel/src/lib.rs:178]: This is the first time I've ever seen `_ if ...` before. Cool!

**Comment by brooksmtownsend** [comment]
This is still a WIP, but I'd like to get this in so we can avoid all the merge conflicts and have an MVP for the telnet channel, any additional comments @autodidaddict ?

**Comment by autodidaddict** [comment]
I think it's fine to continue iterating. We don't need to wait until a single PR has all the telnet-related features.

**Comment by autodidaddict** [comment]
:shipit:

**Pull request merged**